CYCLONEDDS_QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,206 @@
+
+In the source repository, these header files reside with the modules that make up Cyclone DDS.
+
+### API Stability Policy [1.iv]
+
+Cyclone DDS provides Public API stability for PATCH and MINOR releases. Cyclone DDS strives to provide Public API stability for MAJOR releases. Cyclone DDS does not guarantee Public API stability for MAJOR releases.
+
+Cyclone DDS provides Evolving API stability for PATCH releases and strives to provide Evolving API stability for MINOR releases. Cyclone DDS does not guarantee Evolving API stability for MINOR and MAJOR releases.
+
+The RMW interface is what ROS 2 deals with, and for a variety of reasons it may be decided to rely on unstable interfaces (or even implementation details), to better support ROS 2's design decisions that do not fit so well with DDS. Given ROS 2's importance to Cyclone DDS the Public API is expected to eventually cover all of ROS 2's needs, but without a defined time-line for that to happen.
+
+Of course the Cyclone DDS projects takes these uses into account, but it may at times even require source changes to the RMW layer. As the Cyclone DDS project is actively involved in maintaining the RMW layer, such cases are handled by scheduling updates and performing additional compatibility checks against the ROS 2 test suite to ensure they do not cause any disruption to ROS 2.
+
+Those changes are all hidden in the interface between the RMW layer and Cyclone DDS itself and are of no concern to ROS 2 applications. It just means that, in the general case, one should always use a matched pair of `rmw_cyclonedds_cpp` and `cyclonedds`.
+
+### ABI Stability Policy [1.v]
+
+Cyclone DDS provides ABI stability for PATCH releases and strives to provide ABI stability for MINOR releases, for both Public and Evolving APIs. Cyclone DDS does not guarantee ABI stability for MINOR or MAJOR releases.
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+
+ROS 2 users do not interact directly with Cyclone DDS and the mapping of the RMW interface to the Cyclone DDS APIs provided by the `rmw_cyclonedds` package hides any API or ABI changes from ROS 2 users.
+
+Based on the ABI Stability Policy, Cyclone DDS PATCH releases can be accepted as updates within a Released ROS Distribution without also requiring an update to the `rmw_cyclonedds` package. Cyclone DDS MINOR and MAJOR releases can, at least in principle, be accepted as updates within a Released ROS Distribution if the update is accompanied by an update of the `rmw_cyclonedds` package.
+
+## Change Control [2]
+
+Cyclone DDS follows the recommended guidelines of the [Eclipse Development Process](https://www.eclipse.org/projects/dev_process/).
+
+The Eclipse Foundation carefully manages write access to project repositories, allowing only designated [Committers](https://www.eclipse.org/projects/dev_process/#4_1_Committers), who have been voted for in elections overseen by the Eclipse Foundation, to commit changes.
```

Comment:
Nit: I'm not really sure 'carefully' adds any useful information here, and in my opinion, comes across as an effort to make the process appear more robust than it is.

```suggestion
The Eclipse Foundation manages write access to project repositories, allowing only designated [Committers](https://www.eclipse.org/projects/dev_process/#4_1_Committers), who have been voted for in elections overseen by the Eclipse Foundation, to commit changes.
```